ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp

```diff
@@ -426,7 +440,7 @@ void process_shaders(std::vector<std::future<void>>& tasks) {
     }));
 }
 
-void write_output_files() {
+void write_output_files(void) {
```

Comment:
Why would not having `void` in the empty parameter list cause a compiler warning? AFAIK that's required in C, but not C++.

Reply:
C was my "first" real programming language and it's mostly out of habit.

<details>
<summary>here's why by example</summary>

```sh
21:50:05 | ~/Local/code/c
  λ cat test.c 
#include <stdio.h>

void print_text();

void print_text() {
    fprintf(stderr, "Hello, world!\n");
}

int main() {
    print_text();
    return 0;
}
21:50:09 | ~/Local/code/c
  λ gcc -Wall -Wextra -std=c89 -pedantic -Wmissing-prototypes -Wstrict-prototypes test.c -o test
test.c:3:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    3 | void print_text();
      | ^~~~
test.c:5:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    5 | void print_text() {
      |      ^~~~~~~~~~
test.c:9:5: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
    9 | int main() {
      |     ^~~~
21:50:13 | ~/Local/code/c
  λ vim test.c
21:51:23 | ~/Local/code/c
  λ cat test.c
#include <stdio.h>

void print_text(void);

void print_text(void) {
    fprintf(stderr, "Hello, world!\n");
}

int main(void) {
    print_text();
    return 0;
}
21:51:25 | ~/Local/code/c
  λ gcc -Wall -Wextra -std=c89 -pedantic -Wmissing-prototypes -Wstrict-prototypes test.c -o test
21:51:28 | ~/Local/code/c
  λ ./test 
Hello, world!
```
</details>

Reply:
Strict prototypes definitely make sense in C, but in C++ this is already enforced as part of the language: an empty parameter list does actually mean no parameters. For that reason I would rather leave out the `void` here.

Reply:
Ok. Done.

Reply:
Thank you.